Add handling for hr tags

jongracecox opened the pull request:
Add explicit handling of hr tags. This resloves #13, which is an issue due to the incorrect assumption that all tags beginning with "h" and 2 characters long are header tags.

Added a regex for detecting header tags.

Add a method of rendering horizontal rules based on the following github post: https://github.com/python-openxml/python-docx/issues/105#issuecomment-62806373

Add a unittest for horizontal rule handling.

Closes #13.

pqzx commented on htmldocx/h2d.py:
```diff
@@ -315,7 +340,31 @@ def handle_starttag(self, tag, attrs):
+            self.paragraph = self.doc.add_paragraph()
+            pPr = self.paragraph._p.get_or_add_pPr()
+            pBdr = OxmlElement('w:pBdr')
+            insert_element_before(pPr, pBdr, successors=(
```
`insert_element_before()` is now included in python-docx ([this comment](https://github.com/python-openxml/python-docx/issues/105#issuecomment-442786431) in the referenced thread).

jongracecox replied:
I updated the code to use `pPr.insert_element_before` as mentioned in the comment.

pqzx commented on htmldocx/h2d.py:
```diff
@@ -30,6 +32,29 @@
+            return child
+    return None
+
+def insert_element_before(parent, elm, successors):
```
This is supported by python-docx now, so we don't need to add these helper methods.

jongracecox replied:
I removed both of the helper functions.

pqzx reviewed (changes requested):
Thanks a lot for finding a solution for issue #13 and making a PR. Looks good - minor change requested.

jongracecox commented:
I didn't realise that all of the test suite functions write to the same document and that the document is saved during teardown, so I removed the `self.doc.save()` from the new `test_handling_hr()` function I added. This keeps the test in line with the existing tests.
